edg/parts/Ina219.py

```diff
@@ -0,0 +1,77 @@
+from edg import *
+from edg.parts.JlcPart import JlcPart
+
+
+class Ina219_Device(InternalSubcircuit, JlcPart, FootprintBlock, Block):
+    @init_in_parent
+    def __init__(self):
+        super().__init__()
+
+        self.vs = self.Port(VoltageSink(
+            voltage_limits=(3, 5.5) * Volt,
+            current_draw=(6 * uAmp, 1 * mAmp)
+        ))
+        self.gnd = self.Port(Ground())
+
+        dio_model = DigitalBidir.from_supply(
+            self.gnd, self.vs,
+            voltage_limit_tolerance=(-0.3, 0.3) * Volt,
+            input_threshold_factor=(0.3, 0.7)
+        )
+        dio_sda_model = DigitalBidir.from_supply(
+            self.gnd, self.vs,
+            voltage_limit_abs=(-0.3 * Volt, 6.0 * Volt),
+            input_threshold_factor=(0.3, 0.7)
+        )
+
+        self.i2c = self.Port(I2cTarget(DigitalBidir.empty(), addresses=[0x40]))
+        self.i2c.sda.init_from(dio_sda_model)
+        self.i2c.scl.init_from(DigitalSink.from_bidir(dio_model))
```

Comment:
```suggestion
        self.i2c = self.Port(I2cTarget(DigitalBidir.empty(), addresses=[0x40]))
        self.i2c.sda.init_from(DigitalBidir.from_supply(
            self.gnd, self.vs,
            voltage_limit_abs=(-0.3, 6.0) * Volt,
            input_threshold_factor=(0.3, 0.7)
        ))
        self.i2c.scl.init_from(DigitalSink.from_supply(
            self.gnd, self.vs,
            voltage_limit_tolerance=(-0.3, 0.3) * Volt,
            input_threshold_factor=(0.3, 0.7)
        ))
```

Consider just inlining both models, since they're not shared with anything else.

It would make sense to have a shared model on IN+/IN- if you'd like, though.

Reply:
added